zebra/zebra_vrf.c

```diff
@@ -545,7 +546,15 @@ struct route_table *zebra_vrf_other_route_table(afi_t afi, uint32_t table_id,
 	    && (table_id != zebrad.rtm_table_default)) {
 		return zebra_ns_get_table(zns, zvrf, table_id, afi);
 	}
-
+	if (vrf_id != VRF_DEFAULT && table_id != RT_TABLE_MAIN &&
+	    table_id != zebrad.rtm_table_default &&
+	    (zvrf->table_id == RT_TABLE_MAIN ||
+	     zvrf->table_id == zebrad.rtm_table_default)) {
```

Comment:
All these checks are incredibly hard to follow. Usually when things get this complicated it's a sign that the code needs to be refactored. I won't block this PR because of this, but I have this feeling that complexity is growing out of control. At some point we'll need to step back and reorganize this part of the code before adding more stuff on top of it.

Reply:
I simplified the code here.
actually, a table identifier can be set if the table identifier is not used somewhere else.
That is to say here that the VRF must not be a vrf lite.
in other cases, this can be used.